cpp/example/test.c

```diff
@@ -85,8 +79,32 @@ int main(){
    b4.data = (unsigned char*)malloc(37*sizeof(unsigned char));
    
    printf("storage size of st2 is %d \n",storage_size(st2));
+
+   void* subrange = subrange_new(st2, 0 , UINT64_MAX);
+   if (subrange == NULL){
+      perror("failed to init subrange");
```

Comment:
Should it end immediately in case of an error?
```suggestion
      perror("failed to init subrange");
      return;
```